### PR TITLE
wr-151: Link Comments BE with comments for Article

### PR DESCRIPTION
**Pull request by drMangust**
- [x] Empowered both input and Link components.
- [x] Added a new relation to articles.
- [x] Adapted the Article layers to accommodate relations.
- [x] Adapted the comment layers to accommodate relations.
- [x] Implemented "commentsApi" package.
- [x] Implemented the display of the actual number of comments
- [x] implemented a link to the comment section of the selected article
- [x] Created a comment create form and added a comments list to the article page.
- [x] Created actions for comments.

image
image
image
image

### CODE REVIEW

File: backend/src/packages/articles/libs/helpers/get-comments-count-query.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { type QueryBuilder } from 'objection';
+
+import { DatabaseTableName } from '~/libs/packages/database/database.js';
+import { type ArticleModel } from '~/packages/articles/articles.js';
+import { type CommentModel } from '~/packages/comments/comment.model.js';
+
+const getCommentsCountQuery = (
```
Comment (andanf-e) [inline backend/src/packages/articles/libs/helpers/get-comments-count-query.ts]: Let's make this helper as a private method in article repository

File: frontend/src/packages/comments/comments-api.package.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  public async update(
+    payload: CommentUpdateRequestDto,
+  ): Promise<CommentWithRelationsResponseDto> {
+    const response = await this.load(
+      this.getFullEndpoint(CommentsApiPath.$ID, {}),
+      {
+        method: 'PATCH',
+        contentType: ContentType.JSON,
+        payload: JSON.stringify(payload),
```
Comment (andanf-e) [inline frontend/src/packages/comments/comments-api.package.ts]: On backend in comments controller it looks like 'PATCH' route looks into commentId in route param for update

File: frontend/src/pages/articles/components/article-card/article-card.tsx [hunk 3]
```diff
@@ -162,12 +171,20 @@ const ArticleCard: React.FC<Properties> = ({
...
-            <IconButton
-              iconName="comment"
-              className={styles.footerIcon}
-              label={MOCKED_REACTIONS.comments}
-            />
+          <li className={styles.reaction}>
+            <Link
+              to={{
+                pathname: articleRouteById as typeof AppRoute.ARTICLE,
+                hash: 'comments',
```
Comment (andanf-e) [inline frontend/src/pages/articles/components/article-card/article-card.tsx]: Let's create constant for it and use for link hash and comments section anchor `id`

File: frontend/src/slices/articles/articles.slice.ts [hunk 4]
```diff
@@ -128,6 +143,42 @@ const { reducer, actions, name } = createSlice({
...
+    builder.addCase(fetchAllCommentsToArticle.fulfilled, (state, action) => {
+      state.articleComments = action.payload.items;
+      state.articleCommentsDataStatus = DataStatus.FULFILLED;
+    });
+    builder.addCase(updateComment.fulfilled, (state, action) => {
+      const updatedComment = action.payload;
+
+      state.articleComments = state.articleComments.map((comment) =>
+        comment.id === updatedComment.id ? updatedComment : comment,
+      );
```
Comment (andanf-e) [inline frontend/src/slices/articles/articles.slice.ts]:
```suggestion
      state.articleComments = state.articleComments.map((comment) => {
        return comment.id === updatedComment.id
          ? updatedComment
          : comment
      });
```

**Review by andanf-e — CHANGES_REQUESTED** [review]

File: backend/src/packages/articles/libs/helpers/get-comments-count-query.ts (hunk 1 above)
Comment (drMangust) [reply backend/src/packages/articles/libs/helpers/get-comments-count-query.ts]: done - [fc8d566](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/pull/218/commits/fc8d5663a86c1b63b95a023218e1a11337ec3c56)

File: frontend/src/packages/comments/comments-api.package.ts (hunk 2 above)
Comment (drMangust) [reply frontend/src/packages/comments/comments-api.package.ts]: done - [b0377bc](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/pull/218/commits/b0377bca0981eea0cc93973ff36ef932ae66252f)

File: frontend/src/slices/articles/articles.slice.ts (hunk 4 above)
Comment (drMangust) [reply frontend/src/slices/articles/articles.slice.ts]: done - https://github.com/BinaryStudioAcademy/bsa-2023-writorium/pull/218/commits/b0377bca0981eea0cc93973ff36ef932ae66252f#diff-10853bd8170cf40ed804641b734644c28645950c4cd654041e8653b07df225fd

File: frontend/src/pages/articles/components/article-card/article-card.tsx (hunk 3 above)
Comment (drMangust) [reply frontend/src/pages/articles/components/article-card/article-card.tsx]: done - [fa01c40](https://github.com/BinaryStudioAcademy/bsa-2023-writorium/pull/218/commits/fa01c40519b21cd920bcefc429a38963e44f6419)

**Review by andanf-e — DISMISSED** [review]

**Review by andanf-e — APPROVED** [review]

**Review by NazarKovtun1 — APPROVED** [review]

**Pull request merged**